src/cli/completion.rs

```diff
@@ -0,0 +1,67 @@
+use std::io::Cursor;
+
+use clap_complete::generate;
+use color_eyre::eyre::Result;
+use console::style;
+use indoc::formatdoc;
+use once_cell::sync::Lazy;
+
+use crate::cli::command::Command;
+use crate::cli::Cli;
+use crate::config::Config;
+use crate::output::Output;
+
+/// Generate shell completions
+#[derive(Debug, clap::Args)]
+#[clap(alias = "complete", verbatim_doc_comment, after_long_help = AFTER_LONG_HELP.as_str())]
+pub struct Completion {
+    /// Shell type to generate completions for
+    #[clap()]
+    shell_type: Option<clap_complete::Shell>,
+
+    /// Shell type to generate completions for
+    #[clap(long, short)]
+    shell: Option<clap_complete::Shell>,
+}
+
+impl Command for Completion {
+    fn run(self, _config: Config, out: &mut Output) -> Result<()> {
+        let shell_type = self.shell_type.or(self.shell);
+        let shell_type = match shell_type {
+            Some(clap_complete::Shell::Bash) => clap_complete::Shell::Bash,
+            Some(clap_complete::Shell::Zsh) => clap_complete::Shell::Zsh,
+            Some(clap_complete::Shell::Fish) => clap_complete::Shell::Fish,
+            _ => clap_complete::Shell::Zsh,
+        };
```

Comment:
This should be all you need I think
```suggestion
        let shell_type = self.shell_type.or(self.shell).expect("shell must be specified");
```

Reply:
btw `.expect()` will turn the type from `Option<clap_complete::Shell>` to `clap_complete::Shell`. That is also done in your pattern match but without needing to have the `_ => ` at the bottom.

You could also do it like this:

```rust
        let shell_type = self.shell_type.or(self.shell);
        let shell_type = match shell_type {
            Some(clap_complete::Shell::Bash) => clap_complete::Shell::Bash,
            Some(clap_complete::Shell::Zsh) => clap_complete::Shell::Zsh,
            Some(clap_complete::Shell::Fish) => clap_complete::Shell::Fish,
            None => panic!("no shell provided"),
        };
```

or this:

```rust
        let shell_type = match self.shell_type.or(self.shell) {
            Some(shell) => shell,
            None => panic!("no shell provided"),
        };
```

but I still think `.expect()` is the cleanest method

Reply:
Thanks!